### PR TITLE
doc: openthread: cleanup and align for nRF54L

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
cleanup old documentation parts and add nRF54L where necessary

### CODE REVIEW

File: doc/nrf/protocols/thread/overview/architectures.rst [hunk 1]
```diff
@@ -95,9 +95,9 @@ It has the following disadvantages:
 * Bluetooth LE activity can degrade the connectivity on Thread if not implemented with efficiency in mind.
 
 .. figure:: images/thread_platform_design_multi.svg
-   :alt: Multiprotocol Thread and Bluetooth LE architecture (nRF52)
+   :alt: Multiprotocol Thread and Bluetooth LE architecture (nRF52, nRF54L)
 
-   Multiprotocol Thread and Bluetooth LE architecture on nRF52 Series devices
+   Multiprotocol Thread and Bluetooth LE architecture on nRF52 and nRF54L Series devices
```
Comment (wiba-nordic) [inline doc/nrf/protocols/thread/overview/architectures.rst:100]:
```suggestion
   Multiprotocol Thread and Bluetooth LE architecture on nRF52 Series and nRF54L Series devices
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/thread/overview/architectures.rst [hunk 2]
```diff
@@ -74,14 +74,14 @@ This platform design is suitable for the following development kits:
...
-   :rows: nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns, nrf21540dk_nrf52840
+   :rows: nrf52840dk_nrf52840, nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns, nrf54l15dk_nrf54l15_cpuapp, nrf54l15dk_nrf54l15_cpuapp_ns, nrf21540dk_nrf52840
 
 .. _thread_architectures_designs_soc_designs_multiprotocol:
 
 Single-chip, multiprotocol (SoC)
 ================================
 
-nRF52 and nRF53 Series devices support multiple wireless technologies, including IEEE 802.15.4 and Bluetooth® Low Energy (Bluetooth LE).
+nRF52 Series, and nRF53 Series and nRF54L Series devices support multiple wireless technologies, including IEEE 802.15.4 and Bluetooth® Low Energy (Bluetooth LE).
```
Comment (wiba-nordic) [inline doc/nrf/protocols/thread/overview/architectures.rst:84]:
```suggestion
nRF52 Series, nRF53 Series and nRF54L Series devices support multiple wireless technologies, including IEEE 802.15.4 and Bluetooth® Low Energy (Bluetooth LE).
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by wiba-nordic — APPROVED** [review]

File: doc/nrf/protocols/thread/overview/supported_features.rst [hunk 3]
```diff
@@ -124,15 +124,6 @@ The Thread Domain operational configuration enables Thread Devices to join and p
 A user or network administrator may use functions of either Thread Commissioning or Thread Border Routers to set up a common Thread Domain operational configuration for Thread Devices.
 The Thread Devices can belong to different Thread networks or `Partitions <Thread Partitions_>`_ that have potentially different per-network credentials.
 
-.. _ug_thread_12_support_limitations:
-
-Limitations for Thread 1.2 support
```
[the remark this answers is not in the document]
Comment (maciejbaczmanski) [reply doc/nrf/protocols/thread/overview/supported_features.rst:129]: we stopped supporting nrf52833 in CLI a while ago, so the limitation regarding 1.2 and multiprotocol should have been removed back then

**Review by greg-fer — APPROVED** [review]

File: doc/nrf/protocols/thread/overview/architectures.rst [hunk 4]
```diff
@@ -110,7 +110,7 @@ This platform design is suitable for the following development kits:
 
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
-   :rows: nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns
+   :rows: nrf52840dk_nrf52840, nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns, nrf54l15dk_nrf54l15_cpuapp, nrf54l15dk_nrf54l15_cpuapp_ns
```
[the remark this answers is not in the document]
Comment (maciejbaczmanski) [reply doc/nrf/protocols/thread/overview/architectures.rst]: this is an automated table generation (`/includes/sample_board_rows.txt`, looks like that: 
image
) hence underscores and `nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns` as it adds both TFM and non-TFM rows. actually, added `nrf54l15dk_nrf54l15_cpuapp_and_cpuapp_ns` row . @wiba-nordic could you take a look if that's correct?

File: doc/nrf/protocols/thread/overview/architectures.rst [hunk 5]
```diff
@@ -74,14 +74,14 @@ This platform design is suitable for the following development kits:
 
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
-   :rows: nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns, nrf21540dk_nrf52840
+   :rows: nrf52840dk_nrf52840, nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns, nrf54l15dk_nrf54l15_cpuapp, nrf54l15dk_nrf54l15_cpuapp_ns, nrf21540dk_nrf52840
```
[the remark this answers is not in the document]
Comment (maciejbaczmanski) [reply doc/nrf/protocols/thread/overview/architectures.rst]: adressed above

File: doc/nrf/protocols/thread/overview/architectures.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (wiba-nordic) [reply doc/nrf/protocols/thread/overview/architectures.rst]: Looks correct to my eyes, but I recommend checking the doc build once it finishes.

**Review by ArekBalysNordic — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**